textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -64,22 +64,14 @@
     private int resultCursor = -1;
     private int limit = Integer.MAX_VALUE;
     private int offset = 0;
-
+    
     /**
-     * This constructor is used to set the operators whose output is to be
-     * compared and joined and the predicate which specifies the fields and
-     * constraints over which join happens.
+     * Constructs a Join operator using a predicate which specifies the fields and
+     *   constraints over which join happens.
      * 
-     * @param outer
-     *            is the outer operator producing the tuples
-     * @param inner
-     *            is the inner operator producing the tuples
      * @param joinPredicate
-     *            is the predicate over which the join is made
      */
-    public Join(IOperator outerOperator, IOperator innerOperator, IJoinPredicate joinPredicate) {
-        this.outerOperator = outerOperator;
-        this.innerOperator = innerOperator;
+    public Join(IJoinPredicate joinPredicate) {
```

Comment:
The old constructor (which takes 3 parameters: innerOperator, outerOperator, and predicate) is replaced with the new constructor (which takes only the predicate).
The inner and outer operator needs to be specified explicitly using `setInnerInputOperator` and `setOuterInputOperator`. This convention is consistent among all operators.